### PR TITLE
#172499349 Add multi language support

### PR DESCRIPTION
**Pull request by T2Wil**
#### What does this PR do?
Enables a user to switch between the languages for a better experience.

#### Description of Task to be completed?
- Auto detect the preferred language of the user based on the browser settings.
- Add support for English language.
- Add support for Ikinyarwanda language.
- Add support for French language.

#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to have the repo locally.
- run `cd /knights-bn-frontend.git` to get inside the project directory.
- run `git checkout -t origin/ft-multi-language-support-172499349` to switch to this branch.
- run `npm i` to install all dependencies.
- run `npm dev-start` to start the server.
- the browser should start by itself to the homePage.

#### What are the relevant pivotal tracker stories?
[link](https://www.pivotaltracker.com/story/show/172499349)

#### Screenshots (if appropriate)

image

## English version
image

## Ikinyarwanda version
image

## French version
image

### CODE REVIEW

**Comment by EugeneMunya** [comment]
Screen Shot 2020-05-15 at 17 24 04
Put these languages  in a select input

**Review by Kagorora — COMMENTED** [review]
@william, Good Job

Here are my feedback

1. Some lines are not tested
Screen Shot 2020-05-15 at 18 51 44


2. Could you find a way to include those languages in a drop-down
   
Screen Shot 2020-05-15 at 18 51 00

3. Remove commented lines

**Comment by T2Wil** [comment]
> Screen Shot 2020-05-15 at 17 24 04
> 
> Put these languages in a select input

Please @EugeneMunya, fetch to see the changes.

**Comment by T2Wil** [comment]
> @william, Good Job
> 
> Here are my feedback
> 
> 1. Some lines are not tested
> 
> Screen Shot 2020-05-15 at 18 51 44
> 
> 1. Could you find a way to include those languages in a drop-down
> 
> Screen Shot 2020-05-15 at 18 51 00
> 
> 1. Remove commented lines

Thank you @Kagorora , 
please fetch to see all the changes.

**Review by EugeneMunya — APPROVED** [review]

**Review by Kagorora — APPROVED** [review]
Good Job, Everything works as expected

File: src/redux/reducers/errorHandler.js [hunk 1]
```diff
@@ -12,8 +9,8 @@ export default (state = initialState, action) => {
       };
     }
     case CLEAR_ERROR_MSG: {
```
Comment (kagabof) [inline src/redux/reducers/errorHandler.js]: There is no braces after the case in as switch statement
```
switch(action.type){
  case type:
     ...
  default:
     ....
}
```

File: src/redux/reducers/eventHandler.js [hunk 2]
```diff
@@ -78,7 +62,22 @@ export default (state = initialState, action) => {
       };
     }
     case RESET_INPUT_FIELDS: {
-      return initialState;
+      return {
```
Comment (kagabof) [inline src/redux/reducers/eventHandler.js]: This is a a duplication please find a way to solve it.

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/redux/reducers/eventHandler.js (hunk 2 above)
Comment (T2Wil) [reply src/redux/reducers/eventHandler.js]: @kagabof , this is resolved!

File: src/redux/reducers/errorHandler.js (hunk 1 above)
Comment (T2Wil) [reply src/redux/reducers/errorHandler.js]: Resolved!

**Review by kagabof — APPROVED** [review]
Well done.

**Pull request merged**